kernel/src/client/arrow_expression.rs

```diff
@@ -163,8 +163,11 @@ fn evaluate_expression(
                 .iter()
                 .zip(schema.fields())
                 .map(|(expr, field)| evaluate_expression(expr, batch, Some(field.data_type())));
+            let mut fields = output_schema.all_fields();
```

Comment:
this code obviously won't merge. was just a way to test via inducing an error, and I put it here in case anyone wants to play with this PR locally as part of reviewing.